### PR TITLE
fix(container/forms): dirty state on form reset

### PR DESCRIPTION
**Pull request by acateland**
**What is the problem this PR is trying to solve?**
dirty state is not set properly on reset, staying to dirty true

when testing fix in streams i find out that when click a reverse button that goes just after the click on a disable state, the button was in a disable and focus state and those button didn't render properly under those conditions
image

**What is the chosen solution to this problem?**
on reset disaptch a dirty false to the redux store and also update local state

fix button rendering on disabled/focused state

**Please check if the PR fulfills these requirements**

* [x] The PR commit message follows our [guidelines](https://github.com/talend/tools/blob/master/tools-root-github/CONTRIBUTING.md)
* [x] Tests for the changes have been added (for bug fixes / features) And [non reg](./screenshots.md) done before need review
* [ ] Docs have been added / updated (for bug fixes / features)
* [ ] Related design / discussions / pages (not in jira), if any, are all linked or available in the PR

**[ ] This PR introduces a breaking change**

### CODE REVIEW

File: packages/containers/src/ComponentForm/ComponentForm.component.js [hunk 1]
```diff
@@ -172,9 +172,13 @@ export class TCompForm extends React.Component {
...
 				.set('jsonSchema', this.props.state.getIn(['initialState', 'jsonSchema']))
 				.set('uiSchema', this.props.state.getIn(['initialState', 'uiSchema']))
-				.set('properties', this.props.state.getIn(['initialState', 'properties'])),
+				.set('properties', this.props.state.getIn(['initialState', 'properties']))
+				.set('dirty', false),
 		);
-		this.setState({ properties: this.props.state.getIn(['initialState', 'properties']).toJS() });
+		this.setState({
+			properties: this.props.state.getIn(['initialState', 'properties']).toJS(),
+			dirty: false,
```
Comment (romainseb) [inline packages/containers/src/ComponentForm/ComponentForm.component.js]: is dirty in the local state ever used in component form ?

Comment (acateland) [reply packages/containers/src/ComponentForm/ComponentForm.component.js]: it is never used actually indeed

File: packages/theme/src/theme/_visual-helpers.scss [hunk 2]
```diff
@@ -55,6 +55,20 @@
...
+		&:disabled,
+		&.disabled {
+			&:focus,
+			&:hover {
+				background-color: transparent;
+				color: $btn-color;
+				box-shadow: none;
+				border: $border-width solid $btn-color;
+				line-height: $btn-line-height - (2 * $border-width);
+				opacity: 0.4;
```
[the remark this answers is not in the document]
Comment (acateland) [reply packages/theme/src/theme/_visual-helpers.scss]: without rest, the button change on look when being disabled to being disabled and focused
image
image

[the remark this answers is not in the document]
Comment (jsomsanith) [reply packages/theme/src/theme/_visual-helpers.scss]: Seems that only the border and background color are necessary
(example on primary disabled button)
capture d ecran 2018-12-05 a 16 46 01
capture d ecran 2018-12-05 a 16 45 52

**Pull request merged**